[Mia6111] 개인 미션 제출

mia6111 opened the pull request:
윤미현입니다 :-) 진행사항이 미비하네요 ㅠㅠ 주말동안 완료해보겠습니다

mia6111 committed:
개인 미션

javajigi reviewed (commented):
미션 진행하느라 수고했어요.

crongro reviewed (commented):
수고 많으셨어요! 미아가 누구시죠? ㅎ

mia6111 committed:
Add 장바구니 추가 기능 테스트 진행

장바구니 추가 기능 테스트 진행

Commit:
비회원 장바구니 추가 테스트 및 서비스 구현

Commit:
Add 할인율 계산 로직 추가, 테스트

CartProductTest 추가
PriceCalculator - 배송비 포함, 할인 가격 구하는 로직 추가

Commit:
Add 장바구니 담기CartProduct 중복 문제, 반찬 상세페이지 이벤트 처리

- CartProduct 중복 저장 문제 임시 해결
- 반찬 상세 페이지 클릭 이벤트 처리, 애니메이션 적용중

Commit:
Add 제품 상세 페이지 장바구니 담기 프론트 완료

-product.js, cart.js 로 분리, 모듈 사용
-URI 정책 변경 products/categoryId/productId 및 html 링크 변경

Commit:
Update 상세페이지 카트 추가 백엔드 버그픽스 및 프론트 완료

- 카트 추가시생기는 백엔드 버그 픽스 (productCnt 새로 할당되지 않는 문제)
- 프론트 > 카트 추가시 토스터 트랜지션
- 프론트 > 초기 화면 장바구니 툴팁 렌더링

Commit:
Add 반찬 목록 장바구니 담기

- 담기 버튼 누를때 이벤트 핸들링
- 수량 증감시 이벤트 핸들링

Commit:
Refactor 카트 관련 모듈 리팩토링

- 중복 함수 분리 및 함수 쪼개기

Commit:
Add 장바구니 화면 프론트

Commit:
Add 장바구니 화면 변경 요청 프론트

- 프론트 갯수 변경 리스너 부착
- 프론트, handlebar로 장바구니 리스트 출력
- 백엔드 구현중...

mia6111 committed:
Fix 장바구니 중복 문제 및 totalPrice (JPA 이벤트리스터 및 세션 재설정) 버그 픽스

mia6111 committed:
Fix 로그인 후 장바구니 변경 버그 픽스 - Cart의 주인 == User 체크

mia6111 committed:
Add 로그인 시 기존  장바구니 불러오기
